### ISSUE TITLE
Custom metric appears not supported when multi-output is used

### ISSUE THREAD

**Issue by hcho3** [dmlc/xgboost#9730]
https://discuss.xgboost.ai/t/inconsistent-number-of-samples-in-sklearn-with-xgboost/3407

**Comment by trivialfis** [comment]
We just added custom objective support, metric can take a while as the current DMatrix interface cannot return label as matrix
